ui/main/src/app/modules/useractionlogs/useractionlogs.component.ts

```diff
@@ -176,6 +181,39 @@ export class UserActionLogsComponent implements OnInit {
         }
     }
 
+    export(): void {
+        this.initExportData();
+    }
+
+    // EXPORT TO EXCEL
+    initExportData(): void {
+        const exportUserActionLogsData = [];
+
+        const dateColumnName = this.userActionLogsPage.columnTitle.date;
+        const actionColumnName = this.userActionLogsPage.columnTitle.action;
+        const loginColumnName = this.userActionLogsPage.columnTitle.login;
+        const entitiesColumnName = this.userActionLogsPage.columnTitle.entities;
+        const cardUidColumnName = this.userActionLogsPage.columnTitle.cardUid;
+        const commentColumnName = this.userActionLogsPage.columnTitle.comment;
+
+        this.setViewParametersFromForm(0);
+        let userActionLogsForExportResult: UserActionLogsResult = null;
+        this.userActionLogsView.search(true).subscribe((result) => {
+            userActionLogsForExportResult = result;
+            userActionLogsForExportResult.data.content.forEach((line: UserActionLogLine) => {
+                exportUserActionLogsData.push({
+                    [dateColumnName]: line.date,
+                    [actionColumnName]: line.action,
+                    [loginColumnName]: line.login,
+                    [entitiesColumnName]: line.entities,
+                    [cardUidColumnName]: line.cardUid,
+                    [commentColumnName]: line.comment
+                });
+            });
+            ExcelExport.exportJsonToExcelFile(exportUserActionLogsData, 'UserActionLogs');
+        });
+    }
```

Comment:
Put this code which is not angular related in the userActionLogs.view.ts 
No unit test 

I do not think you need to call this.setViewParametersFromForm(0); because the export shall be based on what is visible on screen i.e the last search

Reply:
Done.